pkg/collector/python/datadog_agent.go

```diff
@@ -58,6 +59,18 @@ func GetHostname(hostname **C.char) {
 	*hostname = TrackedCString(goHostname)
 }
 
+// GetHostTags exposes the tags of the agent host to Python checks.
+//
+//export GetHostTags
+func GetHostTags(hostTags **C.char) {
+	tags := hosttags.Get(context.Background(), true, config.Datadog())
```

Comment:
does it make sense to set a timeout here? is it possible collecting these tags can take a while? we probably don't want the agent integration init to just hang forever waiting for this to return (if that's possible)

Reply:
good point! I think it's safest to add a timeout. I noticed the [Get](https://github.com/DataDog/datadog-agent/blob/f370d436b6f6e3d80e8868f68c7db2f6d89ad0cf/comp/metadata/host/hostimpl/hosttags/tags.go#L102) function that we're calling doesn't return an error so we wouldn't have a way of knowing if the context timed out.

There are a few options I can think of to solve that:
1. Update that function to return an error and update the places that call it 
2. I could rename this function `MustGet` and add a `Get` function that does return an error and call that one
> is it possible collecting these tags can take a while?
3. I don't know the answer to this. I'll investigate today and if the answer ends up being no then maybe we don't need a timeout at all.

I'm going to look into how long it takes to collect tags regardless, but right now I'm leaning toward option 2. What do you think?

Reply:
Just to add my two cents here. 

Tags are fetched at startup and the cache is populated. As far as I understand (from reading the code) [the cache is never expired, bypassed or invalidated. ](https://github.com/DataDog/datadog-agent/blob/ca8c80b7c199ef543e14a907a10b0cd7c0e34dee/comp/metadata/host/hostimpl/hosttags/tags.go#L103)

So as long as: 
1. you pass `cache: true` to `Get` (which you are), 
2. you are not calling `Get` for the very first time

this function should always return immediately and never block.

Reply:
oh nice, i didn't notice this [code block](https://github.com/DataDog/datadog-agent/blob/ca8c80b7c199ef543e14a907a10b0cd7c0e34dee/comp/metadata/host/hostimpl/hosttags/tags.go#L103-L108) when i looked at the function originally. That makes sense, so need to change anything @natashadada 👍 

I am still curious about this though:
> also, should we be worried about non utf-8 compliant data? should we sanitize the tag list to make sure that json marshaling won't fail bc of a bad tag?

if a tag set somehow contains bad values we will fail to return any tags. Maybe this is fine and customers can debug based on logs?

Reply:
> I'm also not sure there's a good way to go through each tag and identify bad values.

you could check that each tag value is valid UTF-8. I'm not sure if this happens other places in the pipeline. 

failing for all tags && just doing this log line seems fine to me though:

> log.Warnf("Error getting host tags: %v. Invalid tags: %v", err, tags)

Reply:
ok yeah I agree, I'll update the error message and that should be good (and we can always adjust later if we need to). Thanks for looking through all of this 🙏